codegen/src/api/storage.rs

```diff
@@ -115,26 +115,58 @@ fn generate_storage_entry_fns(
                             (field_name, field_type)
                         })
                         .collect::<Vec<_>>();
-                    // toddo: [AJ] use unzip here?
-                    let tuple_struct_fields =
-                        fields.iter().map(|(_, field_type)| field_type);
-                    let field_names = fields.iter().map(|(field_name, _)| field_name);
+
+                    let field_names = fields.iter().map(|(n,_)| n);
+                    let field_types = fields.iter().map(|(_,t)| t);
+
                     let entry_struct = quote! {
-                        pub struct #entry_struct_ident( #( pub #tuple_struct_fields ),* );
+                        pub struct #entry_struct_ident( #( pub #field_types ),* );
                     };
                     let constructor =
                         quote!( #entry_struct_ident( #( #field_names ),* ) );
-                    let keys = (0..tuple.fields().len()).into_iter().zip(hashers).map(
-                        |(field, hasher)| {
-                            let index = syn::Index::from(field);
-                            quote!( ::subxt::StorageMapKey::new(&self.#index, #hasher) )
-                        },
-                    );
-                    let key_impl = quote! {
-                        ::subxt::StorageEntryKey::Map(
-                            vec![ #( #keys ),* ]
+
+                    let key_impl = if hashers.len() == fields.len() {
+                        // If the number of hashers matches the number of fields, we're dealing with a
+                        // StorageNMap, and each field should be hashed separately according to the
```

Comment:
Comment not strictly accurate, as it would be possible (although useless) to have a single tuple key e.g. `(KeyType,)` for a regular `StorageMap`

Reply:
I tweaked the comment to "something shaped like a StorageNMap"; does that make more sense?

I guess in the 1-key version it doesn't really matter what branch we follow as both have the same effect anyway.